server/app/services/IdentifierBasedPaginationSpec.java

```diff
@@ -0,0 +1,38 @@
+package services;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Optional;
+
+/**
+ * Specifies pagination behavior for a query using identifier-based offset pagination. An offset
+ * identifier identifies the last item of the previous page of results using its sort order
+ * attribute. The type of the identifier {@code T} depends on the type of the underlying attribute
+ * used for sort order.
+ */
+public class IdentifierBasedPaginationSpec<T> {
+
+  public static IdentifierBasedPaginationSpec<Long> MAX_PAGE_SIZE_SPEC_LONG =
+      new IdentifierBasedPaginationSpec<>(Integer.MAX_VALUE);
+
+  private final int pageSize;
+  private final Optional<T> currentPageOffsetIdentifier;
+
+  public IdentifierBasedPaginationSpec(int pageSize) {
+    this.pageSize = pageSize;
+    this.currentPageOffsetIdentifier = Optional.empty();
+  }
+
+  public IdentifierBasedPaginationSpec(int pageSize, Optional<T> currentPageOffsetIdentifier) {
```

Comment:
suggestion: Requiring an optional wrapper seems odd here given the other ctor exists.  I'd remove it unless there's a compelling use I don't see.

Reply:
good call, removed